### ISSUE TITLE
Works with USB power, not with battery?

### ISSUE THREAD

**Issue by codepope** [sqfmi/badgy#20]
Set up badgy on USB power, things behave as expected. Board LED glows flickers blue. Board boots up.

Popped a LIR2450 in with no USB connection and... nothing. Won't wake up, won't reset for new sketches, no LEDs glowing.

Pop in USB with LIR2450 in place, board LED glows bright blue (charging?) and board functions.

Any suggestions as to trouble shooting this?

**Comment by chunkysteveo** [comment]
Just received my new (WHITE PCB) version and can echo the same here - Works fine on USB, appears to charge when an LIR2450 is in place (blue LED) and USB plugged in. Can switch on with an LIR2450 and USB connected, and interestingly when I disconnect the USB while still on it still works. Switch off and back on with just the LIR2450 and no power.. Have not looked into testing where power getting to, and also just testing another LIR2450, but this is odd behaviour indeed!  It's like the USB connection trips it on, and without that initial USB trigger, it doesn't power up?

@codepope  - if your board is working with both USB and LIR connected, disconnect the USB and does it still work on the battery? And then when you switch off and on, does it then fail to boot up?

Would be worth people advising which boards they have, I.E. I now have a blue PCB version with programming headers, and now a super cool white PCB with a serial-USB programming chip on the USB (very cool additon!)

**Comment by codepope** [comment]
That's *exactly* what I'm seeing. (White PCB too). USB+LIR boots, pull the USB, it carries on running, then switch off and back on and it fails to boot up.

**Comment by chunkysteveo** [comment]
hmmmm, maybe the power regulator is not detecting the battery voltage to switch on? Trying another battery here, but sounds like a valid problem.

**Comment by codepope** [comment]
Seems to be another symptom - when you do the switch on with USB and then pull the USB the default sketch reports multiple incorrect directions selected when the stick is touched. Plug the USB back in and it reports as normal.

**Comment by chunkysteveo** [comment]
Yes(!), did not notice that, but now also seeing that the button message reports randomly when going from USB to BATT, and it tends to report multiple button presses, and incorrect ones. It's actually reporting multiple presses as I type this and I am not touching it. Sounds like the button is not being de-bounced correctly, or something to do with the ground - unless on USB which may be grounding things correctly?? Could also tie in with the battery issue, maybe there is an incorrect ground issue when running on battery, so it won't boot up from battery?

**Comment by chunkysteveo** [comment]
@sqfmi - any thoughts on this odd behaviour of the new boards? Noticed the use a new charge IC too, the TP4054.

**Comment by Flet** [comment]
Same exact issue here :( 

New white PCB. Tried two different fully charged batteries.

@sqfmi any thoughts?

**Comment by chunkysteveo** [comment]
Oh ho.... Huston, we have a problem..! 😣

**Comment by sqfmi** [comment]
Sounds like it could be a hardware issue with the new batch, let us investigate the issue and get back to you guys shortly.

**Comment by chadvavra** [comment]
I just received 2 white PCB units and can also confirm that they do not boot, or even power the esp on battery power.

**Comment by sqfmi** [comment]
Brief update, it seems while on battery power only, you are able to turn it on by resetting it. This can be done by using a metal object to short the reset pin with the metal shield (e.g. paperclip, key, etc.):

![image](https://user-images.githubusercontent.com/28943617/46586325-e600c180-ca4a-11e8-86b8-fa6c4096e4c5.png)

Hacky solution for now, will continue to diagnose the root cause.

**Comment by codepope** [comment]
Doesn’t seem to be working for me.

**Comment by chunkysteveo** [comment]
I can confirm this hack works for me and boots the ESP8266 on battery power. 

A better connection to "tap" @codepope would be the **lower** pad of the Lipo JST connection (which is ground). Not the upper pad as that is the positive to BATT. The casing didn't seem to give a good reliable ground connection, but turn on the switch, put a wire (I used a dupoint breadboard wire) on the RESET pin imaged above, and then tap the lower JST pad - the ESP should flash it's blue light on boot. (don't tap the left side pads of the JST connection, these don't seem to be tied to ground, just mechanical fixings)

**Comment by codepope** [comment]
@chunkysteveo Yes, the JST pad is easier to tap, but still no joy. To ensure I was getting a reset, I applied USB power and ground tapping the reset line did reset.

**Comment by chunkysteveo** [comment]
hmmm, not good. Assume battery is fully charged too? Some flake-y issues going on to be sure! Hopefully there will be a fix soon, even if it's a bodge wire fix...!?

**Comment by codepope** [comment]
Well, I thought it was, but testing says otherwise. Tried another battery which shows charged and it didn't play ether.

**Comment by bmv437** [comment]
As a temporary fix, I added a reset button between the ground pad on the JST connector, and the RST pin on the ESP8266. Also threw on a JST connector and LiPo battery since I didn't have a proper coin-cell size LiPo.

Now I just power it on, and click the reset button to boot the ESP8266.

![dpcdmrmxoaapgms](https://user-images.githubusercontent.com/1403707/46694010-871d8280-cbd9-11e8-89fc-0fd5b531d806.jpg)

**Comment by chadvavra** [comment]
I put in a LIR2045 and tried the bodge wire with no luck.  It does work with the USB power though.

**Comment by sqfmi** [comment]
@Korenchkin yes! Thank you for the reminder :) The new schematic is in the hardware folder now.

Everyone else, sorry we're still trying to debug the issue :(

**Comment by Korenchkin** [comment]
thanks for the schematic!i tried to find the problem,but this makes my head hurt..everything seems okay for starting,i'd try pulling the usb chip and stick with prog header(i'm guessing some leakage to ground),but maybe somebody will make smarter solution..like cap on reset?tried that,but did not work for me..but at least i know what the switch switches...

**Comment by sqfmi** [comment]
Removing the CH340C does seem to work! But the reason is unclear at this point... we'll try to do some further investigation with a scope.

**Comment by chadvavra** [comment]
I would like to go on the record as saying that the CH340 is the devil's brood.

I flipping hate those things

**Comment by bmv437** [comment]
@sqfmi If you're looking for cheap alternatives to the CH340, I'm a big fan of the SiLabs CP2104. Adafruit uses them for all their newer dev boards. High data transfer rates, better driver support than the CH340, and cheaper than an FTDI.
Adafruit has more details on their product page: https://www.adafruit.com/product/3309

**Comment by Korenchkin** [comment]
Just a theory,could the ch340 be pulling down gpio0 and/or reset over the 2 transistors?maybe only on start befor it semi-charges its vcc cap?then there are only rx/tx pins,they should not do anything high or low,and that seems all the influence it can do powered off

**Comment by ibuildrockets** [comment]
@sqfmi Hmm.. Reset didn't work for me - the LED on the ESP8266 blinks when reset - but nothing. In fact, I get absolutely nothing happenng now :( sketches won't upload and the currently loaded sketch (weather) won't refresh :,( Oh, and the circled component is scortching hot - 
![20181017_215502](https://user-images.githubusercontent.com/9946962/47083319-f2abc380-d25b-11e8-8b5f-1b476e979080.jpg)

**Comment by chunkysteveo** [comment]
Can you upload a sketch via the USB 2 serial?

**Comment by g2rHP7prvJqO** [comment]
> Removing the CH340C does seem to work! But the reason is unclear at this point... we'll try to do some further investigation with a scope.

I am also suffering from this major issue which makes the Badgy basically 99% useless.
I'm strongly considering de-soldering the CH340C but is there any other active/passive component I need to de-solder too to completely fix this issue and avoid any side effects?
Could you please confirm that I can still flash my badgy using a USB-serial adapter connected to pins 1/3 (first two pins top-right opposite of reset pin) once done with the CH340C?

**Comment by g2rHP7prvJqO** [comment]
> Yes(!), did not notice that, but now also seeing that the button message reports randomly when going from USB to BATT, and it tends to report multiple button presses, and incorrect ones. It's actually reporting multiple presses as I type this and I am not touching it. Sounds like the button is not being de-bounced correctly, or something to do with the ground - unless on USB which may be grounding things correctly??

I observed that too. It has to do with the fact that the UP/DOWN buttons use the same pins as the Serial TX/RX.

**Comment by ibuildrockets** [comment]
> Can you upload a sketch via the USB 2 serial?

No :(

**Comment by codepope** [comment]
Any update @sqfmi ?

**Comment by ibuildrockets** [comment]
@Korenchkin esptool run - I've finally managed to get it to the point where I can upload the hello sketch - however the display is frozen still on the "weather" sketch :(

**Comment by ibuildrockets** [comment]
@Korenchkin I've been using the USB power from the PC, not battery :(

**Comment by pastorhudson** [comment]
Just tried my white badgy on battery for the first time, and nothing. It does work when I plug and unplug the usb cable. But power cycling with the switch doesn't work without usb power.

I also experienced phantom Left button and down button presses. They are consistent. I can't use those functions in sketches because they will override other inputs.

**Comment by sqfmi** [comment]
Sorry for the delay, here's an update to the issue:

1. Though we're still not exactly sure why the CH340C is causing Badgy to malfunction while on battery power only, removing it does resolve the issue. One option is to remove the CH340C and rely solely on OTA to upload your sketches, like in the original Badgy (all code samples currently have this built-in). If your Badgy requires a hardware flash for recovery, refer to the pin-out diagram below:

![image](https://user-images.githubusercontent.com/28943617/47620955-f692dd80-dac6-11e8-9b56-6ad82b1a72e5.png)

2. As per suggestion by @bmv437 on the CP2104, we tried replacing the CH340C with it and it appears to work fine in our week of testing. We created an adapter board on OSH Park [https://oshpark.com/shared_projects/f7gRYBxi](https://oshpark.com/shared_projects/f7gRYBxi) and it can be ordered for about $0.60. See pics below:

![image](https://user-images.githubusercontent.com/28943617/47621047-56d64f00-dac8-11e8-99d8-189fb8df5cbb.png)

![image](https://user-images.githubusercontent.com/28943617/47621058-6e153c80-dac8-11e8-8927-428278501a2e.png)


We will continue to investigate if there are other viable solutions. In the meantime, we've asked the factory to build some adapter boards for us; when they are ready, anyone experiencing this issue may request a board from us. Going forward, all new Badgys will be using the CP2104 (╯°□°）╯︵ ┻━┻

**Comment by chadvavra** [comment]
I found that if you don't want to fully remove the CH340C you can just cut the VCC leg (with a nail clipper in my case) and it will perform as if it's removed.

I figured that if you found another fix it would be easier to bridge that one pin later on.

Huzzah on the adapter boards though.  Those look lovely, and like something I will order a few of for all the other stuff that comes with those dodgy chips.

**Comment by pastorhudson** [comment]
Does the CP2104 also address the button issue? Or should I open a separate issue?

**Comment by sqfmi** [comment]
@ibuildrockets none of the components should be scorching hot, something may be faulty or damaged for that to occur. Did this always occur, or only after uploading the weather sketch?

@pastorhudson following up on #27 @chunkysteveo has created.

**Comment by codepope** [comment]
@pastorhudson taking the ch340c off resolves the button issue (but unfortunately, end of line with my badgy as the epaper apparently cracked in the process.... so can't dig any further).... now to wait for some cp2104 based badgys....

**Comment by chunkysteveo** [comment]
@sqfmi - Is there a way to disable or cut the power to the CH340C when on BATT? Would that allow the CH340C to power up on USB and work, but on BATT be powerless and not interfere with the buttons and also maybe boot the ESP8622 chip up? Are the VUSB and VBATT connected, am looking at the schematic and assume no, using the mosfet as a power switch? Or is that the issue, that it is powerless on BATT and causing the issues?

Love the idea of the CP2104 adaptor board and will be keen to get one, but looking to see if we can salvage the CH340C with some bodge wire somehow...?!

**Comment by chunkysteveo** [comment]
Or how about bodging on a small slide switch to power off the CH340C? From the comment above - cutting the VCC leg of the chip makes the Badgy work. So could we bodge a "PROG/RUN" switch to essentially power off the CH340C chip to allow it to run? Could we not tie this in with the mosfet that controls when the Badgy is running on USB or BATT power?

Steve

**Comment by chadvavra** [comment]
@codepope you can replace the eink really easily.  I changed mine out with a yellow 3 color display.

https://www.waveshare.com/catalogsearch/result/?q=2.9+Inch+E-Paper

**Comment by JonPrevost** [comment]
Received 10 white boards all with this battery problem, haven't looked at the button press issue yet. I'm looking into a capacitor (low pass) solution. If effective I'll post my solution.

**Comment by codepope** [comment]
@chadvavra Already done that and back up and running today. 

@sqfmi any idea when the adapter board (and next batch of badgys) will be about?

**Comment by ibuildrockets** [comment]
@sqfmi - the issue occurred after uploading the weather sketch and resetting the ESP8266 as described in the 'Hacky solution' from October 8th above :(

> Brief update, it seems while on battery power only, you are able to turn it on by resetting it. This can be done by using a metal object to short the reset pin with the metal shield (e.g. paperclip, key, etc.):
> 
> ![image](https://user-images.githubusercontent.com/28943617/46586325-e600c180-ca4a-11e8-86b8-fa6c4096e4c5.png)
> 
> Hacky solution for now, will continue to diagnose the root cause.

**Comment by chunkysteveo** [comment]
Any updates on the diagnostics, the CP "hack" PCB's, any other fixes, revisions of the PCB @sqfmi ?

**Comment by sqfmi** [comment]
Sorry about the delay, we've had trouble finding a fab that could mass produce the CP2014 adapter boards for us. We're in discussion right now with a promising vendor at the moment. We'll keep everyone posted on any new updates and an ETA when it becomes available.

**Comment by chunkysteveo** [comment]
That's good, was worried that the project had been abandoned. Do you have the board files available (the add-on board) for anyone that would like to make the CP2104 themselves?

**Comment by chunkysteveo** [comment]
Brilliant, i'm sure it will help a lot of guys out. I may even try and solder one up...!

**Comment by chunkysteveo** [comment]
Only just spotted your link to the original board project on Oshpark, I could have used that link, apologies!! 

Just out of interest, does anyone know the difference between the CP2104-F03-QFN and the CP2104-QFN? Assume the F03 is a new version, as the "non" F03 chips are cheaper?

**Comment by ibuildrockets** [comment]
I've just ordered the PCB from OHS Park - What's the difference between CP2104-F03-**GMR** and CP2104-F03-**GM** - does it matter? Just wanting to get the IC on the way, too :)

**Comment by sqfmi** [comment]
They are the same part, the difference is just in packaging (**R** stands for reel)

**Comment by jcaron23** [comment]
Hi all,

Are the boards currently sold on Tindie still suffering from this issue, or have they been updated?

Thanks!

**Comment by sqfmi** [comment]
@jcaron23 all new boards will come with the CP2104 instead of the CH340C, which does not have the battery issue.

FYI we found a fab to create the adapter boards for us, we are now waiting for the CP2104s to arrive so we can begin testing it out. If they work ok, we will be offering them to those who were affected by the issue.

![rescueboards](https://user-images.githubusercontent.com/28943617/53568418-f4ea2c80-3b2f-11e9-9f25-39b0566806fc.jpg)

**Comment by chunkysteveo** [comment]
Awesome news!!

**Comment by MarquisdeGeek** [comment]
Is there any alternate fix for those of us with the old board, as shorting the reset pin didn't work for me?

**Comment by chunkysteveo** [comment]
What's the difference between the  CP2104-**F03**-QFN and the CP2104-QFN?

@MarquisdeGeek - you can snip the VCC pin of the CH chip to disable it. You will lose the ability to program it via USB though.

**Comment by MarquisdeGeek** [comment]
@chunkysteveo - Thanks, but that's a bit drastic for me!

**Comment by emanuelstanciu** [comment]
@sqfmi Hi, I have one of the blue PCB models which was bought late last year and I suffer from exactly the same behaviour as described on this post. Any idea what's happening?
Works fine on USB power, but on Battery simply flashes the ESP blue LED and the nothing happens.
Any pointers are greatly appreciated

**Comment by chunkysteveo** [comment]
@emanuelstanciu  -  I would recommend starting a fresh issue with this problem. This issue is related to the USB to serial chip not allowing the ESP8266 to boot when on battery. The blue PCB version does not have any USB to serial chip, so whatever issue you are seeing, it isn't to do with this issue. I don't want your issue to be lost in this bigger post!

If the blue light flashes, that would indicate that it has booted and is powered. Something else is not working here?

Steve

**Comment by pingywon** [comment]
Any resolution here? I just got my badgy and was under the impression that the white REV B board did not suffer from this problem. I just got some fresh batteries and Badgy doesnt work at all on battery power.

Can someone please advise.

**Comment by sqfmi** [comment]
Hi @pingywon, are you using LIR2450 batteries?

**Comment by pingywon** [comment]
Not exactly. I have CR2450s

**Comment by sqfmi** [comment]
You must use the LIR2450. The CR2450 is not rechargeable, and is also not the right voltage (3.0V), Check out the README for more details.

**Comment by pingywon** [comment]
OK Great. So My original assumption that the issue has been resolved in my Badgy is correct?

**Comment by chunkysteveo** [comment]
Yeah, REV B does not suffer the power issue on battery when using a. LIR2450 rechargeable battery.

**Comment by Carltk** [comment]
What ever happened to the offer from March 1 about the CP2104 adaptor boards. Do we have to do 
anything to register our interest? I have 2 of the affected boards.

**Comment by sqfmi** [comment]
Hey @Carltk, send us a message with your Tindie order number and we'll ship the adapter boards to your mailing address. Thanks!

**Comment by Springwald** [comment]
I also have 2 badgys with this issue. Any chance to get the adapter boards? Thanks!

**Comment by sqfmi** [comment]
Hey @Springwald, send us a message with your Tindie order number and we'll ship the adapter boards to your mailing address. Thanks!

**Comment by MarquisdeGeek** [comment]
What is the preferred channel to send the message, since I also need an adapter board? Thanks.

**Comment by sqfmi** [comment]
Hey @MarquisdeGeek , reach out to us via Tindie with your order number, and we'll ship the adapter boards to your mailing address. Thanks!
